aya/src/programs/links.rs

```diff
@@ -1,19 +1,54 @@
 use libc::{close, dup};
+
 use std::{
+    borrow::Borrow,
     collections::{hash_map::Entry, HashMap},
+    ops::Deref,
     os::unix::prelude::RawFd,
 };
 
 use crate::{generated::bpf_attach_type, programs::ProgramError, sys::bpf_prog_detach};
 
-pub(crate) trait Link: std::fmt::Debug + 'static {
+/// A Link
+pub trait Link: std::fmt::Debug + 'static {
+    /// Unique Id
     type Id: std::fmt::Debug + std::hash::Hash + Eq + PartialEq;
 
+    /// Returns the Link UUID
```

Comment:
I'd change this to say 

```
Returns the link id.
```

I'm actually thinking that we should split this in two traits, where `detach` would be public and `id` wouldn't. `id` is an internal method really I don't think it makes sense to expose it. But happy to do that in a followup PR if you prefer (I expect it's probably going to be a bit of a chore).